Ensuring the collector behavior remains consistent when mTLS feature gate is enabled but TA is not deployed

ItielOlenick opened the pull request:
**Description:** 
Ensuring the collector behavior remains consistent when mTLS feature gate is enabled but TA is not deployed

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #3456

**Testing:** 
Local

**Documentation:** 
No documentation changes needed

jaronoff97 commented:
Some test failures, but i think this is a fine fix for now.

ItielOlenick commented:
Pretty sure the CI test failure has nothing to do with the code changes.

swiatekm commented:
> Pretty sure the CI test failure has nothing to do with the code changes.

It looks like the only unit tests failing are ones related to the mTLS feature, so that would be quite the coincidence.

swiatekm commented:
@ItielOlenick could you add a changelog entry? I wouldn't mind seeing a simple e2e test with the flag enabled but `targetAllocator` disabled.

ItielOlenick commented:
@swiatekm anything missing?

swiatekm commented:
> @swiatekm anything missing?

You said you'd add an e2e test, but I don't see it in this PR.

ItielOlenick commented:
> > @swiatekm anything missing?
> 
> You said you'd add an e2e test, but I don't see it in this PR.

Sorry, i misread that for tests in general.
Added both
For some reason the changelog (which I added) is not picked up in the recent CI job.

swiatekm commented on tests/e2e-ta-collector-mtls/ta-disabled/00-install.yaml:
```diff
@@ -0,0 +1,76 @@
+  - kind: ServiceAccount
+    name: collector
+    namespace: ($namespace)
+---
```
The ClusterRole, binding and Service here aren't necessary, I don't think.

swiatekm commented on tests/e2e-ta-collector-mtls/ta-disabled/00-install.yaml:
```diff
@@ -0,0 +1,76 @@
+  name: prometheus-cr
+spec:
+  config: |
+    receivers:
```
To make this test more targeted, I think we should use a prometheus receiver here. The same config as in https://github.com/open-telemetry/opentelemetry-operator/blob/01abcc32a148217e323217eb06dd521e08ddaf74/tests/e2e/smoke-targetallocator/00-install.yaml#L35 should work.

swiatekm commented:
> For some reason the changelog (which I added) is not picked up in the recent CI job.

Can you rebase your changes on main? Github doesn't show this in the diff, but your branch has changelog fragments from before the most recent release.

ItielOlenick commented:
> > For some reason the changelog (which I added) is not picked up in the recent CI job.
> 
> Can you rebase your changes on main? Github doesn't show this in the diff, but your branch has changelog fragments from before the most recent release.

Done

swiatekm commented:
The changelog issue is fixed in https://github.com/open-telemetry/opentelemetry-operator/pull/3595, so you won't need to delete the entry file after that's merged.
